### ISSUE TITLE
Discuss the issue of CI testing for LSX instructions

### ISSUE THREAD

**Issue by loongson-zn** [numpy/numpy#25288]
https://github.com/numpy/numpy/pull/25215#issue-2004051415

Dear @rgommers @seiko2plus, about LSX instruction CI testing problem, I did some work.
First, ubuntu does not yet have a suitable toolchain, so ci (https://github.com/numpy/numpy/blob/main/.github/workflows/linux_qemu.yml) is not applicable now.
Second， I have verified through [LoongArch sbuildQEMU on debian community](https://wiki.debian.org/LoongArch/sbuildQEMU) .  I have stored some software gcc and qemu that need to be upgraded( supporting simd), [Click here](https://github.com/loongson-zn/qemu-debian)  have gcc and qemu binary,  rootfs can running on  x86. If you want to do it yourself, you can refer to [README.md](https://github.com/loongson-zn/qemu-debian/blob/master/README.md)
If both of you have other ideas, I would be happy to help.

Particularize:
[gcc master branch](https://gcc.gnu.org/git/gcc.git) and [qemu master branch](https://github.com/qemu/qemu.git)  already supports the extension vector instruction of LoongArch(LSX).

**Comment by loongson-zn** [comment]
If you have debian os(maybe other system is ok) , simple methods:
```
$ wget https://github.com/loongson-zn/qemu-debian/blob/master/qemu/qemu-loongarch64-static
$ cp qemu-loongarch64-static /usr/bin/qemu-loongarch64-static
$ cat >/tmp/qemu-loongarch64 <<EOF
package qemu-user-static
type magic
offset 0
magic \x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x02\x01                    
mask \xff\xff\xff\xff\xff\xff\xff\xfc\x00\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff
interpreter /usr/bin/qemu-loongarch64-static
EOF
$ update-binfmts --import /tmp/qemu-loongarch64 

$ wget https://github.com/loongson-zn/qemu-debian/blob/master/la-rootfs.tar.gz
$ tar -xvf la-rootfs.tar.gz
$ cd srv/chroots/sid-loong64-sbuild/root/
$ cp /usr/bin/qemu-loongarch64-static  srv/chroots/sid-loong64-sbuild/usr/bin
$ chroot .
$ mount -t proc proc proc
$ mount -t sysfs sys sys
$ mount -t devtmpfs dev dev 
$ mount -t devpts devpts dev/pts 
$ mount -t tmpfs shmfs dev/shm
``` 
then, start `spin test`

**Comment by loongson-zn** [comment]
> In that case, feel free to create an independent workflow limited to LoongArch based on Debian.

Yes, but I don't know how to submit the PR related to numpy CI testing.

**Comment by loongson-zn** [comment]
I enjoyed doing it

**Comment by seiko2plus** [comment]
> I enjoyed doing it

Sounds great!

**Comment by charris** [comment]
I think this is done.

**Comment by seiko2plus** [comment]
> I think this is done.

Nice catch, indeed!
